### DISCUSSION TITLE
307 redirect for all _rsc requests

### DISCUSSION THREAD

**Discussion by landvibe** [vercel/next.js#55415] [App Router]
We are adding a locale prefix to the URL for i18n purposes.
When navigating between pages, a 307 redirect is occurring for all _rsc requests.

For example, in the project below, if you create a page called "/test",
https://github.com/vercel/next.js/tree/canary/examples/app-dir-i18n-routing
and add <Link href='/test'>go to test</Link> to the root page,
when you click the button, a request for /test?_rsc=8ndtn is made,
and a 307 redirect to /en/test occurs.

I'm wondering if this is the expected behavior when using i18n with the app router.

**Comment by PovarovDenis** [comment]
I have almost the same issue. I have custom logic in `middleware.ts` related to redirects. And as a side effect I have also redirects for all `?_rsc=...` requests. In general I don't care about it but I concern about another thing - after redirect the request don't have already the `?_rsc=...` query parameter and being cached by CDN as a page. The CDN issue I bypass using special cache rules in the CDN (bypass cache for all pages where header Rsc=1).

<img width="793" alt="image" src="https://github.com/user-attachments/assets/1bc6f6da-0994-4363-988f-a02090a4c6e1">
